OOInteraction/src/commands/CAddCalleesToView.cpp

```diff
@@ -0,0 +1,114 @@
+ ** DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ ** SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ ** SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ ** WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ ** OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ **
+ **********************************************************************************************************************/
+
+#include "CAddCalleesToView.h"
+#include "VisualizationBase/src/items/ViewItem.h"
+#include "OOModel/src/declarations/Method.h"
+#include "OOModel/src/expressions/MethodCallExpression.h"
+
+
+namespace OOInteraction {
+
+CAddCalleesToView::CAddCalleesToView()
+	:CommandWithDefaultArguments("addCallees", {"current"})
+{
+}
+
+bool CAddCalleesToView::canInterpret(Visualization::Item* source, Visualization::Item* target,
+	const QStringList& commandTokens, const std::unique_ptr<Visualization::Cursor>& cursor)
+{
+	bool canInterpret = CommandWithDefaultArguments::canInterpret(source, target, commandTokens, cursor);
+	while (source && !source->hasNode())
+		source = source->parent();
+	if (!source)
+		return false;
+	else
+		return canInterpret && dynamic_cast<OOModel::Method*>(source->node());
+}
+
+Interaction::CommandResult* CAddCalleesToView::executeWithArguments(Visualization::Item* source, Visualization::Item*,
+		const QStringList& arguments, const std::unique_ptr<Visualization::Cursor>&)
+{
+	while (source && !source->hasNode())
+		source = source->parent();
+	auto name = arguments.at(0);
+
+	if (name == "current")
+		name = source->scene()->currentViewItem()->name();
+	auto view = source->scene()->viewItem(name);
+
+	if (view)
+	{
+		auto callees = getCallees(source->node());
+		auto pos = view->positionOfNode(source->node());
+
+		if (callees.size() > 0)
+		{
+			//TODO@cyril What if it is in the view, but not as a top-level item?
+			if (pos.x() == -1)
+			{
+				view->insertColumn(0);
+				view->insertNode(source->node(), 0, 0);
+				pos = view->positionOfNode(source->node());
+			}
+			//TODO@cyril Insert the nodes in a good order
+			view->insertColumn(pos.x() + 1);
+			auto row = callees.size() - 1;
+			for (auto callee : callees)
+				view->insertNode(callee, pos.x() + 1, row--);
+		}
+		return new Interaction::CommandResult();
+	}
+	else
+		return new Interaction::CommandResult(new Interaction::CommandError("View " + name + " does not exist"));
+}
+
+QSet<OOModel::Method*> CAddCalleesToView::getCallees(Model::Node* parent)
+{
+	QSet<OOModel::Method*> result;
+	for (auto child : parent->children())
+	{
+		if (auto call = dynamic_cast<OOModel::MethodCallExpression*>(child))
```

Comment:
This finds all the callees of the parent node, which is instantiated with the method. It only gets the callees of the method that the command is executed on.